node/src/components/consensus/protocols/highway/synchronizer/tests.rs

```diff
@@ -183,12 +180,21 @@ fn do_not_download_synchronized_dependencies() {
         &peer0,
         HighwayMessage::RequestDependency(Dependency::Unit(c0)),
     );
+    // `c1` is now part of the synchronizer's state, we should not try requesting it if other
+    // vertices depend on it.
+    assert!(matches!(
+        *sync.schedule_add_vertex(peer0, pvv(b0), now),
+        [ProtocolOutcome::QueueAction(ACTION_ID_VERTEX)]
+    ));
     // "Download" the last dependency.
     let _ = sync.schedule_add_vertex(peer0, pvv(c0), now);
     // Now, the whole chain can be added to the protocol state.
-    for unit in &[c0, c1, b0, c2] {
+    let units: std::collections::BTreeSet<Dependency<TestContext>> = vec![c0, c1, b0, c2]
+        .into_iter()
+        .map(Dependency::Unit)
+        .collect();
+    for _ in 0..units.len() {
         let (maybe_pv, outcomes) = sync.pop_vertex_to_add(&highway);
```

Comment:
The test would be slightly stronger if you'd instead iterate over the vertices to add:
```rust
while let (Some(pv), outcomes) = sync.pop_vertex_to_add(&highway) {
    // …
    assert!(units.remove(&pv_dep));
    // …
}
```
That way we don't just check that we get at least `units.len()` vertices and they are all contained in `units`, but actually that none of them repeat and they are exactly identical to `units`.

Reply:
After changing the code as you asked in https://github.com/CasperLabs/casper-node/pull/1312#discussion_r614081731, there's only one dependency now being requested.

Reply:
Done in [`60627c6` (#1312)](https://github.com/CasperLabs/casper-node/pull/1312/commits/60627c68a9551e909c18b7ad4107de17aea904a9).